### ISSUE TITLE
Implement minimum cache size for States::minimizeCache

### ISSUE THREAD

**Issue by AustinSanders** [DOI-USGS/ale#525]
States::minimizeCache seems to be over-reducing the size of the cache as described in https://github.com/DOI-USGS/ISIS3/issues/5023.

For reference:

In order to correctly solve for acceleration using jigsaw, the table / cache must contain 3 or more records. Based on discussion that we found in https://github.com/USGS-Astrogeology/ISIS3/issues/3928, ale creates caches that are significantly smaller than those created by ISIS. It appears that, in some cases, ALE may in fact be over-minimizing the cache and abstracting away information that is necessary to solve for accelerations. We may be able to adjust this behavior by configuring the tolerance in [ALE States::minimizeCache](https://github.com/USGS-Astrogeology/ale/blob/main/include/ale/States.h#L126). Alternatively, since adjusting the tolerance doesn't guarantee that we won't over-minimize the cache, it might make sense to come up with a way to specify a minimum cache size.

**Comment by jrcain-usgs** [comment]
From lwellerastro on this/related issues:
The data in the Phobos/jigsaw post was lost when /scratch failed.  The images at the time were processed under an older version of isis (7.0.0) which was incorrectly picking up ale 0.8.6 and applying a no-vetted driver to the MEX data. Under that scenario my post says I had 10 images where CacheType=Linear, not HermiteSpline as expected.  

In the post, AustinSanders calls out [4 images he found with the problem](https://github.com/DOI-USGS/ISIS3/issues/5023#issuecomment-1452263466).  I have since reprocessed my images under isis8.0.0RC-1 and those images did not pick up the ale driver, but the isis driver, so they don't have the problem.  Under the latest isis8.0.0 I ingested and spiced 2 of the images that Austin called out in the post (HG751_0000_GR2 and H3843_0000_ND2) and both are picking up the new ale driver.  But both also have CacheType=HermiteSpline.

So I'm wondering if the new driver has a vetted/good driver for MEX and if the minimumCache size is no longer a problem for these data?

**Comment by lwellerastro** [comment]
After testing one image fairly thoroughly, I think we are doing better with the new ale 0.9.1, at least for MEX HRSC which is the project that highlighted this problem. I have a couple of corrections to my text that Jacob shared above, but only ale version number.

This particular image and the others that had problems are very low res and have very few lines of spiced data (small disk in a sea of space). Not sure that explains anything, but it is typical for flyby data and maybe something to consider with the caching for certain cameras/misions.

Test data under my work area Isis3Tests/Jigsaw/Phobos/MinimumCacheIssues/. See proc.scr for details if that would be useful. 

The problem originally occurred under isis7.0.0, ale 0.8.7 and has the following InstumentPosition info and tabledump output:
```
  Name                 = InstrumentPosition
  StartByte            = 1565471
  Bytes                = 112
  Records              = 2
  ByteOrder            = Lsb
  CacheType            = Linear
  SpkTableStartTime    = 543598628.44577
  SpkTableEndTime      = 543598640.39514
  SpkTableOriginalSize = 2.0
  Description          = "Created by spiceinit"
  Kernels              = $mex/kernels/spk/ORMM__170301000000_01336.BSP

tabledump from=HG751_0000_BL2_sourceAle087.cub name=InstrumentPosition
J2000X,J2000Y,J2000Z,J2000XV,J2000YV,J2000ZV,ET
-2416.4619015198,615.2867873512,-1642.6972812552,1.7645131887515,0.67070174419125,-2.1718990449693,543598628.44577
-2395.3727349627,623.30461557826,-1668.641376041,1.7652347320624,0.6712844547862,-2.1704203805288,543598640.39514
```
Under 8.0.0-RC1 the data do not pick up the ale driver (ale 0.8.6), but pick up the isis driver:
```
  Name                 = InstrumentPosition
  StartByte            = 1565855
  Bytes                = 168
  Records              = 3
  ByteOrder            = Lsb
  CacheType            = HermiteSpline
  SpkTableStartTime    = 543598628.44577
  SpkTableEndTime      = 543598640.39514
  SpkTableOriginalSize = 145.0
  Description          = "Created by spiceinit"
  Kernels              = $mex/kernels/spk/ORMM__170301000000_01336.BSP

tabledump from=HG751_0000_BL2_sourceIsis.cub name=InstrumentPosition
J2000X,J2000Y,J2000Z,J2000XV,J2000YV,J2000ZV,ET
-2416.4619015198,615.2867873512,-1642.6972812552,1.7645131887515,0.67070174419125,-2.1718990449693,543598628.44577
-2405.9184030068,619.29486588945,-1655.6715148364,1.7648764434344,0.67098380594491,-2.1711679111229,543598634.42045
-2395.3727349627,623.30461557826,-1668.641376041,1.7652390694447,0.67126126251121,-2.1704352701868,543598640.39514
```
Under the public release 8.0.0, the image picks up ale 0.9.1 and matches the isis driver results:
```
  Name                 = InstrumentPosition
  StartByte            = 1565855
  Bytes                = 168
  Records              = 3
  ByteOrder            = Lsb
  CacheType            = HermiteSpline
  SpkTableStartTime    = 543598628.44577
  SpkTableEndTime      = 543598640.39514
  SpkTableOriginalSize = 145.0
  Description          = "Created by spiceinit"
  Kernels              = $mex/kernels/spk/ORMM__170301000000_01336.BSP

tabledump from=HG751_0000_BL2_sourceAle091.cub name=InstrumentPosition
J2000X,J2000Y,J2000Z,J2000XV,J2000YV,J2000ZV,ET
-2416.4619015198,615.2867873512,-1642.6972812552,1.7645131887515,0.67070174419125,-2.1718990449693,543598628.44577
-2405.9184035894,619.29486907769,-1655.6715127972,1.7648785870086,0.67097206438881,-2.171175419899,543598634.42045
-2395.3727349627,623.30461557826,-1668.641376041,1.7652390694447,0.67126126251121,-2.1704352701868,543598640.39514
```
This still might be a topic to look out for for future drivers, but I think it's doing what we expect for at least the MEX HRSC images I'm working with. Is there something else we should dig into to verify we're doing ok here?

**Comment by acpaquette** [comment]
@lwellerastro Can this be closed via external communications with intent to reopen if the issue arises again?

**Comment by lwellerastro** [comment]
My last set of results and comments from last week confirm things look good as well. Thanks!

**Comment by acpaquette** [comment]
Just for context if this issue needs to be reopened:

Under ALE 0.8.6 the number of instrument positions extracted from ephemeris data was being reduced too much in ALE (to two positions). This over reduction of data prevented solving of velocities in the jigsaw app in ISIS. The problem was corrected in ALE 0.8.8 and later integrated into ISIS under ALE 0.9.1, if solving for ephemeris values on other drivers become an issue this PR may be reopened.
